src/connector/src/sink/encoder/template.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::catalog::Schema;
+use risingwave_common::row::Row;
+use risingwave_common::types::ToText;
+
+use super::{JsonEncoder, Result, RowEncoder};
+use crate::sink::SinkError;
+
+pub enum RedisFormatEncoder {
```

Comment:
I think we don't need an extra enum for redis. The `RedisFormatterEncoder` has no special redis logic, and `SinkFormatterImpl` is already an enum.